docsrc/imap/developer/compiling.rst

```diff
@@ -112,46 +121,65 @@ Alternate database formats
+    `shapelib`_, shapelib, shapelib, "yes/no", "no", "It is required for
+    **tzdist** service to have geolocation support. Otherwise it is not needed.
+    Version 1.3.0 or higher is required when using it."
+    `wslay`_, libwslay-dev, wslay-devel, "no", "no", "It provides WebSockets
+    support in httpd. Only used with **JMAP**, otherwise not needed."
+    `zlib`_, zlib1g-dev, zlib-devel, "no", "no", "It provides gzip compression
+    support for http communications."
 
 Other
 #####
 
 .. csv-table::
-    :header: "Package", "Debian", "RedHat",  "Required for ``make check``?", "Notes"
-    :widths: 20,15,15,5,45
-
-     SSL certificates, ssl-cert-dev, mod_ssl, "no", "Used if you're installing SSL certificates"
-    `ClamAV`_,,,, Used by **cyr_virusscan**.
-    `net-snmp`_, libsnmp-dev, net-snmp-devel, "no", "version 4.2 or higher"
-    `openldap`_, libldap2-dev, openldap-devel, "no", "Development headers to enable **ptloader** to interface with LDAP
-    directly, for canonification of login usernames to mailbox names,
-    and verification of login usernames, ACL subjects and group
-    membership.
+    :header: "Package", "Debian", "RedHat", "Required", "Required for ``make check``?", "Notes"
+    :widths: 20,15,15,5,5,45
+
+    SSL certificates, ssl-cert-dev, mod_ssl, "no", "no", "Used if you're
+    installing SSL certificates."
+    `ClamAV`_, libclamav-dev, clamav-devel, "no", "no", "It is used by
+    **cyr_virusscan**, otherwise not needed."
+    `openldap`_, libldap2-dev, openldap-devel, "no", "no", "Development headers
+    to enable **ptloader** to interface with LDAP directly, for canonification
+    of login usernames to mailbox names, and verification of login usernames,
+    ACL subjects and group membership.
 
     Configure option: ``--with-ldap``"
-    `tcp_wrappers`_, tcp_wrappers, xx, "no"
-    `transfig`_, transfig, xx, "no", "also known as fig2dev"
-    `pcre`_,,,, "PCRE 1 (or 8) - for regular expression matching"
-    `perl(Term::ReadLine)`_,,,, "Perl library needed by cyradm"
-    `libsrs2`_,,,, "Sender Rewriting Scheme for lmtp, used on Sieve redirect "
-    `zlib`_, zlib1g-dev, zlib-devel, "no", "Compression support for httpd"
-    `libbrotli`_,,,, "Brotli compression support for httpd"
-    `wslay`_,,,, "WebSockets support in httpd"
-    `nghttp2`_, libnghttp2-dev, libnghttp2-devel, "no", "HTTP/2 support for httpd"
+    `pcre`_, libpcre3-dev, pcre-devel, "yes", "no", "PCRE 1 (8.x) - for utf-8/unicode regular expression
+    matching. Also see `issues/1731`_."
+    `perl(Term::ReadLine)`_,,, "no", "no", "Perl library needed by **cyradm**"
+    `libsrs2`_, *no package*, *no package*, "no", "no", "It is used for
+    implementing Sender Rewriting Scheme (SRS) functionality for messages
+    forwarded by sieve scripts. Without it, messages forwarded by sieve scripts
+    will not have this functionality and might have difficulty delivering to
+    SMTP servers that insist on it."
 
```

Comment:
🤷‍♂️ Don't know. I would remove this feature entirely, it looks like it's not used much, complex to setup and it's unclear what servers may require it. The dependency itself is not in the major distros, should be built manually.

Reply:
Presumably the admin installing Cyrus knows what their outgoing SMTP server requires well enough to decide for themselves whether they need this feature or not.  So I think it's fine for us to just say that some SMTP servers may insist on it, without needing to list which ones do under which circumstances.

Reply:
I do not think that a receiving SMTP server can require SRS.  As there is no fixed schema how to do the SRS, a receiving server cannot know, if SRS in involved.  The rewriting is solely in the interest of the sending SMTP server (so that the sending server gets any bounces).